From: Jordan Santell <jsantell@gmail.com>
Date: Wed, 18 Mar 2015 11:50:52 -0700
Subject: Bug 1143915 - Fix test for MemoryActor now that it backs out
 when attempting to restart recording allocations when its already recording.
 a=ryanvm

diff --git a/toolkit/devtools/server/tests/mochitest/test_memory_allocations_04.html b/toolkit/devtools/server/tests/mochitest/test_memory_allocations_04.html
index 0c2e620..5568736 100644
--- a/toolkit/devtools/server/tests/mochitest/test_memory_allocations_04.html
+++ b/toolkit/devtools/server/tests/mochitest/test_memory_allocations_04.html
@@ -29,32 +29,32 @@ window.onload = function() {
 
     var testProbability = Task.async(function* (p, expected) {
       info("probability = " + p);
       yield memory.startRecordingAllocations({
         probability: p
       });
       allocator();
       var response = yield memory.getAllocations();
+      yield memory.stopRecordingAllocations();
       return response.allocations.length;
     });
 
     is((yield testProbability(0.0)), 0,
        "With probability = 0.0, we shouldn't get any allocations.");
 
     ok((yield testProbability(1.0)) >= 100,
        "With probability = 1.0, we should get all 100 allocations (plus "
        + "whatever allocations the actor and SpiderMonkey make).");
 
     // We don't test any other probabilities because the test would be
     // non-deterministic. We don't have a way to control the PRNG like we do in
     // jit-tests
     // (js/src/jit-test/tests/debug/Memory-allocationsSamplingProbability-*.js).
 
-    yield memory.stopRecordingAllocations();
     yield memory.detach();
     destroyServerAndFinish(client);
   });
 };
 </script>
 </pre>
 </body>
 </html>
-- 
2.2.1

